rest/src/main/java/org/jboss/pnc/rest/provider/BuildConfigurationProvider.java

```diff
@@ -255,10 +264,41 @@ public BuildConfigurationAuditedRest getRevision(Integer id, Integer rev) {
         return new BuildConfigurationAuditedRest (auditedBuildConfig);
     }
 
+    public Response getLatestBuildRecord(Integer configId) {
+        BuildConfiguration buildConfiguration = buildConfigurationRepository.queryById(configId);
+        if (buildConfiguration == null) {
+            return Response.status(Response.Status.NOT_FOUND).entity("No build configuration exists with id: " + configId).build();
+        }
+        List<BuildRecord> buildRecords = buildRecordRepository.findAllByLatestBuildConfigurationOrderByEndTimeDesc(buildConfiguration);
```

Comment:
I would recommend using https://github.com/project-ncl/pnc/blob/master/spi/src/main/java/org/jboss/pnc/spi/datastore/repositories/SortInfoProducer.java#L17 with proper arguments. It would look simpler and you will have to write less code.